### DISCUSSION TITLE
Visual representation of multihop enabled

### DISCUSSION THREAD

**Discussion by 666Danger-sudo** [mullvad/mullvadvpn-app#9073] [Feature requests & Ideas]
## Improvment report

Operating system: 
Can be implemented on all multihop supporting versions. 
App version: 

### Improvment description

So when the multihop is enabled, the image on the main page don't show the multihop relay as you can see:

![img1](https://user-images.githubusercontent.com/46642145/232200294-5ff01b88-c4b4-40d7-ab56-44adc3580786.png)

It would be a better representation of the actual connection and would make the user feel more secure if you can implement a visual representation of the multihop like this:

![img2](https://user-images.githubusercontent.com/46642145/232200607-89be5e43-fb77-49f3-8031-5d7eff44e016.png)

**Comment by MrChocolatine** [comment]
I like this suggestion.
Not fundamentally necessary but it would a nice visual addition in terms of user experience.

**Comment by raksooo** [comment]
Thanks for the suggestion! I'm a bit uncertain of how it should display really long distances. Something like this would probably be nice but I don't think we'll prioritize it anytime soon unfortunately, but I'll keep this issue open to keep track of the request.

**Comment by faern** [comment]
It won't be able to show two locations at opposite sides of the globe at once. So it can't be implemented directly as suggested. But I guess we can look at some way of better visualizing it.

I guess you have noticed that the server names say `fi-hel-wg-003 via fr-par-wg-005`, which show you that you do have multihop on and which servers are used. The map and the big country/city labels are just for your *exit*, it shows from what endpoint you talk to the internet.

**Comment by 666Danger-sudo** [comment]
So I have a suggestion for that. What about implementing dynamic map for the gui so you can scroll the glob. This way, you can place two dots at long distance and the user needs to scroll through the glob to see the entry node and exit node OR when multihop enabled, put a miniature 3d glob where the user see the two dot of multihop so he don't needs to scroll. Idk I will let you guys think about that. I think that the gui of mullvad is subject to imrpovments. You cannot keep the same Gui among years it will become boring even if technically it does not matter. Thx
